### PR TITLE
improve wxWebViewIE EmulationLevel switching and offline routine

### PR DESCRIPTION
**Pull request by chilauhe**
### -  MSWSetModernEmulationLevel changes : 
former one can only set to IE8 or IE7, which are not match up with modern Web. 
I made it possible to choose what ever wanted while keeping backward compatibility and safe.
[Ref. Controlling WebBrowser Control Compatibility](https://blogs.msdn.microsoft.com/patricka/2015/01/12/controlling-webbrowser-control-compatibility/)

### - fixed minor problem caused by OfflineMode: 
when you are not connected, IE should open localhost web server but wxWebViewIE would get "Navigate Cancelled" error page. Made it works same behavior as IE to save some poor little programmers' day :-P
fixed random assert error caused by wrong type cast in SetOfflineMode by the way. 

SetOfflineMode function actually died, IE already hidden this option in menu, remove it someday not hurt any.

### CODE REVIEW

File: src/msw/webview_ie.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ bool wxWebViewIE::Create(wxWindow* parent,
     m_container = new wxIEContainer(this, IID_IWebBrowser2, m_webBrowser, m_uiHandler);
 
     EnableControlFeature(21 /* FEATURE_DISABLE_NAVIGATION_SOUNDS */);
-
+    SetOfflineMode(false);
```
Comment (vadz) [inline src/msw/webview_ie.cpp]: It would be nice to have a comment explaining why is this needed.

File: include/wx/msw/webview_ie.h [hunk 2]
```diff
@@ -146,7 +163,11 @@ class WXDLLIMPEXP_WEBVIEW wxWebViewIE : public wxWebView
 
     // Establish sufficiently modern emulation level for the browser control to
     // allow RunScript() to return any kind of values.
-    static bool MSWSetModernEmulationLevel(bool modernLevel = true);
+    static bool MSWSetModernEmulationLevel(wxWebViewIE_EmulationLevel modernLevel = wxWEBVIEWIE_EMU_IE8);
```
Comment (vadz) [inline include/wx/msw/webview_ie.h]: The default parameter should be removed, otherwise there is an ambiguity when just `MSWSetModernEmulationLevel()` is called.

File: include/wx/msw/webview_ie.h [hunk 3]
```diff
@@ -36,6 +36,23 @@ class DocHostUIHandler;
...
+    wxWEBVIEWIE_EMU_DEFAULT = 0,
+    wxWEBVIEWIE_EMU_IE7 = 7000,
+    wxWEBVIEWIE_EMU_IE8 = 8000,
+    wxWEBVIEWIE_EMU_IE8_FORCE = 8888,
+    wxWEBVIEWIE_EMU_IE9 = 9000,
+    wxWEBVIEWIE_EMU_IE9_FORCE = 9999,
+    wxWEBVIEWIE_EMU_IE10 = 10000,
+    wxWEBVIEWIE_EMU_IE10_FORCE = 10001,
+    wxWEBVIEWIE_EMU_IE11 = 11000,
+    wxWEBVIEWIE_EMU_IE11_FORCE = 11001,
```
Comment (vadz) [inline include/wx/msw/webview_ie.h]: The trailing comma here should be removed.

**Review by vadz — COMMENTED** [review]
Thanks for your changes, I have absolutely no objections to them but the new enum/overload need to be documented in `interface/wx/webview.h`, could you please do it?

TIA!

File: include/wx/msw/webview_ie.h (hunk 2 above)
Comment (chilauhe) [reply include/wx/msw/webview_ie.h]: I rename the method to `MSWSetEmulationLevel()` in 2e7a0151e6b72880fd1154f8ef8f39e1adfe2cde  , because "Modern" dose not suitable for ones like IE7 that must older than a decade.

**Comment by chilauhe** [comment]
> Thanks for your changes, I have absolutely no objections to them but the new enum/overload need to be documented in `interface/wx/webview.h`, could you please do it?
> 
> TIA!

Certainly yes, I've made a new commit. If anything not so good please let me know and refine it.
I really appreciate your guidance, maybe yesterday I stay up too late and just a mess in my head that time😂.
